tools/cldr/allscripts.sh

```diff
@@ -35,8 +35,10 @@ node genlikelyloc.js $TARGET
 # this uses genlikelyloc's output, so it has to come after it
 node genlang2charset.js $TARGET
 node genlist.js $TARGET
-
+node genmeasurementsystem.js $TARGET
 node gennumfmt.js $TARGET
 node genpapersizes.js $TARGET
+# node genplurals.js $TARGET   plurals need a rework before this can be uncommented
```

Comment:
Yes, the whole plural support is not working the way I wanted it. There are two ways to make it work, and originally I had one way, and now the JS code has been rewritten to work the other way. Both approaches work, but one is more efficient than the other. The problem is that the Java version of the code has not been rewritten to work the other way, so the code is just plain broken now. If you run the Java unit tests, you can see they are just plain broken. In fact, the build.xml does not run the Java unit tests anymore because of this.

The two ways to choose a choice, given a pivot value and a set of choices from a choice string:

- try each choice in sequence and test if the pivot value is in the class for that choice. If so, use the string for that choice. If all choices fail, use the "other" choice. This is the new approach that only the JS code uses.

- map the pivot value to a class, and then check each choice to see if it has that class. If a choice with that class does not exist, use the "other" choice. This is the old way, and the way that the Java code works right now.

The difference between the two approaches is that the first approach recalculates the containment for every choice every time, whereas the second approach only calculates the class containment once. Basically, "is this value in this class?" versus "give me the class for this value."

What I wanted to do is rewrite both the JS and the Java code to use the same approach and the same data from CLDR, and rewrite the genplurals.js to match.

Reply:
oh I see. I haven't run unit test in Java part  before. only I cared JS part.